src/hotspot/share/oops/arrayOop.hpp

```diff
@@ -104,12 +104,19 @@ class arrayOopDesc : public oopDesc {
 
   // Accessors for instance variable which is not a C++ declared nonstatic
   // field.
-  int length() const {
-    return *(int*)(((intptr_t)this) + length_offset_in_bytes());
+  int length() const { return *length_addr(); }
```

Comment:
Personally, I find the original code easier to read, since the identical content, `*(int*)(((char*)mem) + length_offset_in_bytes())` exists in both `length()` and `set_length(HeapWord* mem, int length)`.

Reply:
I should have included the 2-arg set_length in the changes to make everything consistent.  I'll revise in the next update.